src/ripple/app/paths/impl/BookStep.cpp

```diff
@@ -350,22 +389,23 @@ BookStep<TIn, TOut>::revImp (
         {
             auto ofrAdjAmt = ofrAmt;
             auto stpAdjAmt = stpAmt;
-            limitStepOut (
-                offer.quality (), ofrAdjAmt, stpAdjAmt, transferRateIn, remainingOut);
+            auto ownerGivesAdj = ownerGives;
+            limitStepOut (offer.quality (), ofrAdjAmt, stpAdjAmt, ownerGivesAdj,
+                transferRateIn, transferRateOut, remainingOut);
             remainingOut = beast::zero;
             savedIns.insert (stpAdjAmt.in);
             savedOuts.insert (remainingOut);
             result.in = sum(savedIns);
             result.out = out;
-            this->consumeOffer (sb, offer, ofrAdjAmt, stpAdjAmt);
+            this->consumeOffer (sb, offer, ofrAdjAmt, stpAdjAmt, ownerGivesAdj);
             return false;
         }
     };
 
     {
-        auto const r = forEachOffer<TIn, TOut> (
-            sb, afView, book_,
-            strandSrc_, strandDst_, eachOffer, maxOffersToConsume_, j_);
+        auto const prevStepRedeems = prevStep_ && prevStep_->redeems (sb, false);
```

Comment:
Generally speaking, I dislike functions that accept a boolean to select one of two kinds of behavior. A function is a much better option since the name can document the otherwise obscure boolean input. Compare with:

    prevStep_->redeedsForward(sb);
    prevStep_->redeemsReverse(sb);

Not suggesting you change anything at this time.